### ISSUE TITLE
Installation Problem related to SimpleFileFetcher.php

### ISSUE THREAD

**Issue by KirbyJames** [JeroenDeDauw/Maps#353]
### Setup

- MW version:   1.29.0
- DB (MySQL etc.):   5.6.36
- PHP version:   5.6.31
- Maps version:   4.3
- SMW version (if applicable):   2.5

### Issue

I've installed Mediawiki 1.29.0 and then used Composer to add Maps
```json
 "require": {
        "mediawiki/maps": "~4.3",
	"mediawiki/semantic-media-wiki": "~2.5"
    }
```
When I try to add a simple map e.g.  {{#display_map:Brandenburg Gate, Berlin, Germany}}
I get an error in SimpleFileFetcher.php - see below.

SimpleFileFetcher.php requires PHP > 7  which is why it fails on my installation (PHP = 5.6.31).

I've looked through the tarball for Mediawiki - and it doesn't contain ../vendor/jeroen/file-fetcher/src/SimpleFileFetcher.php

Is SImpleFileFetcher loaded with Maps or SMW?

Any suggestions as to how I can overcome this problem appreciated!

Kirby

-------------------------------------------------

Warning: Unsupported declare 'strict_types' in /home/lkjh/public_html/vendor/jeroen/file-fetcher/src/SimpleFileFetcher.php on line 3

Parse error: syntax error, unexpected ':', expecting ';' or '{' in /home/lkjh/public_html/vendor/jeroen/file-fetcher/src/SimpleFileFetcher.php on line 21

Fatal error: Uncaught exception 'MWException' with message 'Language::isValidBuiltInCode must be passed a string, NULL given' in /home/lkjh/public_html/languages/Language.php:371 Stack trace:

```
#0 /home/lkjh/public_html/includes/cache/localisation/LocalisationCache.php(453): Language::isValidBuiltInCode(NULL)
#1 /home/lkjh/public_html/includes/cache/localisation/LocalisationCache.php(336): LocalisationCache->initLanguage(NULL)
#2 /home/lkjh/public_html/includes/cache/localisation/LocalisationCache.php(273): LocalisationCache->loadItem(NULL, 'namespaceGender...')
#3 /home/lkjh/public_html/languages/Language.php(600): LocalisationCache->getItem(NULL, 'namespaceGender...')
#4 /home/lkjh/public_html/includes/title/MediaWikiTitleCodec.php(86): Language->needsGenderDistinction()
#5 /home/lkjh/public_html/includes/title/MediaWikiTitleCodec.php(217): MediaWikiTitleCodec->getNamespaceName(0, 'SandBoxAdmin')
#6 /home/lkjh/public_html/includes/cache/LinkCache.php(187): MediaWikiTitleCodec->getPrefixedDBkey(Object(Title))
#7 /home/lkjh in /home/lkjh/public_html/languages/Language.php on line 371
```

**Comment by kghbln** [comment]
That's strange. I your environment `jeroen/file-fetcher` 3.1.0 should be installed instead of the latest version as seen on [sandbox](https://sandbox.semantic-mediawiki.org/wiki/Sp%C3%A9cial:Version). However the wiki linked updates from earlier versions though this should not make a difference compared to doing a fresh install like you did.

**Comment by KirbyJames** [comment]
kghbln,

Thanks for your note. I've just changed the composer.local.json file to include

  "require" {
    "jeroen/file-fetcher": "3.1.0"
}
and run composer update

and it now works fine!

Thanks

Kirby

**Comment by kghbln** [comment]
@KirbyJames I still believe that there may be something in the water since you should not necessarily add the file-fetcher to your require section. @JeroenDeDauw Since you are the master of dependency hell. ;)

**Comment by JeroenDeDauw** [comment]
I'm not sure what happened. Indeed a too new version of FileFetcher was installed, which can be fixed by pinning it to 3.1.0. The question is how the too new version was installed though. Composer should prevent that. I double checked the 4.0 release of FileFetcher, and the constraints there are fluff... Perhaps a tarball was created on a PHP 7.x box and then uploaded?

**Comment by kghbln** [comment]
> Perhaps a tarball was created on a PHP 7.x box and then uploaded to a 5.x box?

This indeed currently appears to be the only viable cause for this.
